test/browser/full_page_refresh_test.rb

```diff
@@ -51,4 +51,22 @@ class FullPageRefreshTest < ActionDispatch::IntegrationTest
     page.evaluate_script('window.history.back()')
     assert_equal "Sample Turbograft Application", page.title
   end
+
+  test "tg-static preserves client-side state of innards on partial refresh, and replaces contents on full refresh" do
+    page.fill_in 'badgeinput', :with => 'tg-static innards'
+    click_link "Perform a full page refresh"
+    assert_equal "tg-static innards", find_field("badgeinput").value
+    click_link "Perform a partial page refresh and refresh the navigation section"
+    assert_equal "", find_field("badgeinput").value
```

Comment:
I think this is correct, just a bit nuanced:

When doing a full page refresh, the stuff inside a `tg-static` (the badgeinput innards) should remain and not be swapped out
But when explicitly replacing the element (`"refresh the navigation section"`) we should see a new DOM element appear (hence the badgeinput now being `''`)